src/test/java/plugins/GitLabPluginTest.java

```diff
@@ -0,0 +1,78 @@
+package plugins;
+
+import jakarta.inject.Inject;
+import org.jenkinsci.test.acceptance.docker.DockerContainerHolder;
+import org.jenkinsci.test.acceptance.docker.fixtures.GitLabContainer;
+import org.jenkinsci.test.acceptance.junit.*;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import java.net.http.HttpResponse;
+
+import java.io.IOException;
+
+import static org.junit.Assert.*;
+
+@WithDocker
+@Category(DockerTest.class)
+@WithPlugins("gitlab-plugin")
+@WithCredentials(credentialType = WithCredentials.SSH_USERNAME_PRIVATE_KEY, values = {"gitlabplugin", "/org/jenkinsci/test/acceptance/docker/fixtures/GitLabContainer"})
+public class GitLabPluginTest extends AbstractJUnitTest {
+
+    @Inject
+    DockerContainerHolder<GitLabContainer> gitLabServer;
+
+    private GitLabContainer container;
+    private String repoUrl;
+    private String host;
+    private int port;
+
+    private String privateToken;
+
+    public String getPrivateToken() {
+        return privateToken;
+    }
+
+    @Before
+    public void init() {
+        container = gitLabServer.get();
+        repoUrl = container.getRepoUrl();
+        host = container.host();
+        port = container.port();
+        container.waitForReady(this);
+        
+        try {
+            privateToken = container.createUserToken();
+        } catch (InterruptedException e) {
```

Comment:
Do not convert to `RuntimeExceptions, just throw it and the test will fail if needed